### PR TITLE
Gui.download (#64)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
```<|Download file|button|on_action={lambda s: gui.download("/downloads/fred.png")}|>```

### CODE REVIEW

File: taipy/gui/gui.py [hunk 1]
```diff
@@ -745,11 +766,11 @@ def show_notification(
             type (string): The notification type. This can be one of `"success"`, `"info"`, `"warning"` or `"error"`.
```
Comment (FabienLelaquais) [inline taipy/gui/gui.py]: (optional(string)), right?

**Review by FabienLelaquais — APPROVED** [review]
This looks really, really good!
Still my open question about whether this API (as well as some others) should be on the State... but that's another topic.

File: taipy/gui/gui.py (hunk 1 above)
Comment (FredLL-Avaiga) [reply taipy/gui/gui.py]: yes indeed

**Review by dinhlongviolin1 — APPROVED** [review]
Thanks for the included test :D

**Pull request merged**